catalog/transformer/catalog-transformer-pdf/src/main/java/ddf/catalog/transformer/input/pdf/PdfInputTransformer.java

```diff
@@ -143,78 +149,37 @@ public Metacard transform(InputStream input) throws IOException, CatalogTransfor
-        }
-    }
-
-    private Metacard transformWithExtractors(InputStream input, String id)
-            throws IOException, CatalogTransformerException {
         try (TemporaryFileBackedOutputStream fbos = new TemporaryFileBackedOutputStream()) {
             try {
                 IOUtils.copy(input, fbos);
             } catch (IOException e) {
                 throw new CatalogTransformerException("Could not copy bytes of content message.",
                         e);
             }
-
             ByteSource docByteSource = fbos.asByteSource();
-
-            try (InputStream isCopy = docByteSource.openStream(); //
+            Metacard metacard;
+            try (InputStream isCopy = docByteSource.openStream();
                     PDDocument pdfDocument = pdDocumentGenerator.apply(isCopy)) {
-
-                Parser parser = new AutoDetectParser();
-                try (InputStream metaIs = docByteSource.openStream(); //
-                        TemporaryFileBackedOutputStream contentHandlerStream = new TemporaryFileBackedOutputStream()) {
-                    ContentHandler contentHandler = new ToTextContentHandler(contentHandlerStream,
-                            StandardCharsets.UTF_8.toString());
-                    TikaMetadataExtractor tikaMetadataExtractor = new TikaMetadataExtractor(parser,
-                            contentHandler);
-                    ByteSource plainTextByteSource = extractPlainText(metaIs,
-                            contentHandlerStream,
-                            tikaMetadataExtractor);
-
-                    return transformPdf(id, pdfDocument, plainTextByteSource);
+                try (InputStream metaIs = docByteSource.openStream()) {
+                    metacard = transformPdf(id, pdfDocument, metaIs);
                 }
             } catch (InvalidPasswordException e) {
                 LOGGER.debug("Cannot transform encrypted pdf", e);
                 return initializeMetacard(id);
             }
+            return metacard;
         }
     }
 
-    private ByteSource extractPlainText(InputStream metaIs,
-            TemporaryFileBackedOutputStream contentHandlerStream,
-            TikaMetadataExtractor tikaMetadataExtractor) throws IOException {
-        ByteSource plainTextByteSource = null;
-        try {
-            tikaMetadataExtractor.parseMetadata(metaIs, new ParseContext());
-            plainTextByteSource = contentHandlerStream.asByteSource();
-        } catch (CatalogTransformerException e) {
-            LOGGER.warn("Cannot extract metadata from pdf", e);
-        }
-        return plainTextByteSource;
-    }
-
     private MetacardImpl initializeMetacard(String id) {
-        return initializeMetacard(id, null);
+        return initializeMetacard(id, null, null);
     }
 
-    private MetacardImpl initializeMetacard(String id, ByteSource contentByteSource) {
+    private MetacardImpl initializeMetacard(String id, String bodyText, String metadataXml) {
```

Comment:
This is not acceptable and is removing the file backed protections that were put in place to keep this from blowing up with an out of memory error. Passing in an extremely large file will cause this class to blow up the heap with these changes.

Reply:
@stustison - the bodyText is truncated before this method is called preventing a heap issue.  https://github.com/codice/ddf/pull/2302/files/27acd0e43375d33d4a8d6666c989a737feb119fc#diff-b675468460852adc1ced1e04cd2b41d0R222

Reply:
The default for the extracted text passed is 5000000 bytes currently.